lib/src/local_working_copy.rs

```diff
@@ -549,6 +548,7 @@ impl TreeState {
             file_states: FileStatesMap::new(),
             sparse_patterns: vec![RepoPathBuf::root()],
             own_mtime: MillisSinceEpoch(0),
+            symlink_support: check_symlink_support().unwrap_or(false),
```

Comment:
Perhaps, `get_updated_tree_value()` can also be changed to use this flag?
```rust
            let current_tree_values = current_tree.path_value(repo_path);
            let new_file_type = if cfg!(windows) { // XXX HERE
                let mut new_file_type = new_file_state.file_type.clone();
                if matches!(new_file_type, FileType::Normal { .. })
                    && matches!(current_tree_values.as_normal(), Some(TreeValue::Symlink(_)))
                {
                    new_file_type = FileType::Symlink;
                }
                new_file_type
            } else {
                new_file_state.file_type.clone()
            };
```

Reply:
good catch, i've changed it